website/docs/src/content/cloud/observability.mdx

```diff
@@ -0,0 +1,269 @@
+---
+title: Observability & tracing
+description: Monitor and debug your GenSX workflows
+---
+
+# Observability & tracing
+
+GenSX provides observability tools that make it easy to understand, debug, and optimize your workflows. Every component execution is automatically traced, giving you full visibility into what's happening inside your LLM workflows. You can view traces in realtime as workflows execute, and view historical traces to debug production issues like hallucinations.
+
+## Viewing traces
+
+When you run a workflow, GenSX automatically generates a trace that captures the entire execution flow, including all component inputs, outputs, and timing information.
+
+### Accessing the trace viewer
+
+The GenSX cloud console includes a trace viewer. You can access traces in several ways:
+
+1. **From the Console**: Navigate to your project in the [GenSX Console](https://app.gensx.com) and select the "Executions" tab
+2. **Trace URL**: When running a workflow with `printUrl: true`, a direct link to the trace is printed to the console
+3. **API Response**: The execution ID from API responses can be used to view traces in the console
+
+```tsx
+// Executing a workflow with trace URL printing
+const result = await MyWorkflow.run(
+  { input: "What is GenSX?" },
+  { printUrl: true },
+);
+
+// Console output includes:
+// [GenSX] View execution at: https://app.gensx.com/your_org/executions/your_execution_idf
+```
+
+### Viewing the flame graph
+
+The flame graph visualizes the entire execution tree including branches, all nested sub-components, and timing:
+
+![Workflow component tree](/cloud/trace.png)
+
+- **Component hierarchy**: See the nested structure of your components and their relationships
+- **Execution timing**: The width of each bar represents the relative execution time
+- **Status indicators**: Quickly spot errors or warnings with color coding
+- **Component filtering**: Focus on specific components or component types
+
+Click on any component in the flame graph to inspect its details, including inputs, outputs, and timing information.
+
+### Viewing component inputs and outputs
+
+For each component in your workflow, you can inspect:
+
+1. **Input properties**: All props passed to the component
+2. **Output values**: The data returned by the component
+3. **Execution duration**: How long the component took to execute
+4. **Metadata**: Additional information like token counts for LLM calls
```

Comment:
Is this technically true? token counts are only there if they're in the component output (for example <ChatCompletion> won't have token details)

Reply:
Will leave this as I expect we'll improve it in the near future.